### ISSUE TITLE
False negative on `Style/RedundantParentheses` for single condition in parentheses combined with &&

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#14866]
The cop correctly identifies redundant parentheses when wrapping a logical && expression inside another &&, but fails to detect them if the wrapped expression is simplified to a single condition.

## Steps to reproduce the problem

The code that generates an offense (correct behavior):
```ruby
def self.tokens
  @tokens ||= token_classes.filter_map do |token_klass|
    next if !enabled? && (token_klass != A && token_klass != B)

    [token_klass.token_code, token_klass]
  end.to_h.freeze
end
```

The code that gives a false negative:
```ruby
def self.tokens
  @tokens ||= token_classes.filter_map do |token_klass|
    next if !enabled? && (token_klass != A)

    [token_klass.token_code, token_klass]
  end.to_h.freeze
end
```

## Expected behavior

Both cases should be flagged as redundant parentheses

## RuboCop version

```
$ [bundle exec] rubocop -V
1.84.1 (using Parser 3.3.10.1, Prism 1.8.0, rubocop-ast 1.49.0, analyzing as Ruby 3.4, running on ruby 3.4.8) [x86_64-linux]
  - rubocop-capybara 2.22.1
  - rubocop-factory_bot 2.28.0
  - rubocop-migration 0.7.1
  - rubocop-performance 1.26.1
  - rubocop-rails 2.34.3
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
  - rubocop-rspec_rails 2.32.0
  - rubocop-thread_safety 0.7.3
```

**Comment by FFederi** [comment]
The current behavior seems to be intentional, at least in the second case:
https://github.com/rubocop/rubocop/blob/06b84550f0501bf60113c4ce82322d6ddc647dd0/spec/rubocop/cop/style/redundant_parentheses_spec.rb#L1099-L1101

**Comment by koic** [comment]
The first example, `next if !enabled? && (token_klass != A && token_klass != B)`, is detected by `Style/RedundantParentheses` cop.

As @FFederi mentioned, the second example is intentional behavior and is not a false negative.
